### PR TITLE
executor,ast: check password format for 'create user identified by password XXX'

### PR DESCRIPTION
**Pull request by tiancaiamao**
'create user xxx identified by yyy', yyy can be any text
'create user xxx identified by password yyy', yyy must be a hash string generated by
password(), it begin with * and has length 41

@zz-jason @jackysp @winkyao

### CODE REVIEW

File: ast/misc.go [hunk 1]
```diff
@@ -457,6 +457,28 @@ func (u *UserSpec) SecurityString() string {
...
+		return auth.EncodePassword(opt.AuthString), true
+	}
+
+	if !maybePasswordString(opt.HashString) {
+		return "", false
+	}
+	return opt.HashString, true
+}
+
+func maybePasswordString(str string) bool {
```
Comment (jackysp) [inline ast/misc.go]: The function name is weird :)

Comment (tiancaiamao) [reply ast/misc.go]: So, any suggestion? @jackysp

Comment (zz-jason) [reply ast/misc.go]: how about "isPasswordValid" ?

**Pull request merged**